src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -50,9 +50,19 @@ public final class InlineConfigParser {
     /** A pattern matching the symbol: "\" or "/". */
     private static final Pattern SLASH_PATTERN = Pattern.compile("[\\\\/]");
 
-    /** A pattern to find the string: "// violation". */
+    /** A pattern to match the following comment formats:
+     * <ol>
+     *     <li> "// violation" </li>
+     *     <li> "// violation, 'violation message'" </li>
+     *     <li> "// violation 'violation messages'" </li>
+     * </ol>
+     *
+     * This pattern will not match comments of the form "// violation, explanation" or
+     * "// violation, explanation, 'violation message'". These are matched by
+     * {@link InlineConfigParser#VIOLATION_WITH_EXPLANATION_PATTERN}.
+     */
     private static final Pattern VIOLATION_PATTERN = Pattern
-            .compile(".*//\\s*violation\\s*(?:'(.*)')?$");
+            .compile(".*//\\s*violation,?\\s*(?:'(.*)')?$");
```

Comment:
We have a lot of examples of the form `// violation, 'check message'`, extending the regexp to accommodate this will make it easier for contributors (and I prefer the comma separation myself).

Reply:
I am not sure why this parser cares about anything other than `violation` and content of `'...'` in general or why we needed two different regexps, if anyone knows please share. Otherwise - we should probably just combine the two because we only care about the optional matching group.

Reply:
Small observation - It would be nice to expand other pattern comments like this one, wouldn't it?

Reply:
We can probably overhaul/ consolidate a few regexp in this file, really this was a hack to use the format that I prefer and see a fair amount in existing documentation and tests.